docs/solutions/sql_server_always_on_availability_groups/modules/network/subnet/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "oci_core_subnet" "subnet" {
+  count      = "${var.ad_count}"
+  cidr_block = "${var.cidr_block["${count.index}"]}"
+
+  // Some terraform trickery required here.
+  // Due to the fact that we need to specify placement for the Witness server, we're using additional variable ${ad_placement}
```

Comment:
I'm confused by this. The count is always set to ad_count regardless. Also, isn't 0 the first index into "data.oci_identity_availability_domains.ADs.availability_domains"?

Reply:
I admit this is appears to be unnecessary to me as well.  Since I inherited this, I simply tested for functionality and some consistency, but did not do a stringent code review to look for non-optimized code such as this.  I would like to push this into the refactor list at a later date.